pkg/registry/registry.go

```diff
@@ -146,10 +146,54 @@ func (or OktetoRegistry) HasGlobalPushAccess() (bool, error) {
 	return or.client.HasPushAccess(image)
 }
 
+// GetRegistryAndRepo returns image and registry of a given image
 func (or OktetoRegistry) GetRegistryAndRepo(image string) (string, string) {
 	return or.imageCtrl.GetRegistryAndRepo(image)
 }
 
+// GetRepoNameAndTag returns the repo name and tag of a given image
 func (or OktetoRegistry) GetRepoNameAndTag(repo string) (string, string) {
 	return or.imageCtrl.GetRepoNameAndTag(repo)
 }
+
+// CloneGlobalImageToDev clones an image from the global registry to the dev registry
+func (or OktetoRegistry) CloneGlobalImageToDev(imageWithDigest, tag string) (string, error) {
+	descriptor, err := or.client.GetDescriptor(imageWithDigest)
+	if err != nil {
+		return "", err
+	}
+
+	// parsing the image URI to extract registry and repository name
+	reg, repositoryWithTag := or.imageCtrl.GetRegistryAndRepo(imageWithDigest)
+	repo, _ := or.imageCtrl.GetRepoNameAndTag(repositoryWithTag)
+
+	if !strings.HasPrefix(repo, fmt.Sprintf("%s/", or.imageCtrl.config.GetGlobalNamespace())) {
+		return "", fmt.Errorf("image %s is not in the global registry", repo)
+	}
+
+	// forging a new image URI in the dev registry, using the same repo name and tag as the global image
+	globalNamespacePrefix := fmt.Sprintf("%s/", or.imageCtrl.config.GetGlobalNamespace())
+	personalNamespacePrefix := fmt.Sprintf("%s/", or.config.GetNamespace())
+	devRepo := strings.Replace(repo, globalNamespacePrefix, personalNamespacePrefix, 1)
+	devImage := fmt.Sprintf("%s/%s:%s", reg, devRepo, tag)
+
+	newRef, err := name.ParseReference(devImage)
+	if err != nil {
+		return "", err
+	}
+
+	i, err := descriptor.Image()
+	if err != nil {
+		return "", err
+	}
+
+	// writing the new image to the registry
+	err = or.client.Write(newRef, i)
```

Comment:
What about keeping `GetDescriptor` private, and create a Public client function `WriteImageMetadata` and get the descriptor and image inside this function?

Reply:
I think it's a good idea but I've tried to follow this suggestion and implement it that way, what I found odd is that the `WriteImageMetadata` function would require 2 inputs, one being the `imageWithDigest` which is the global image (to find the descriptor and image object), the second being `newRef` which is the metadata of the new image. 

I will leave it as is for now, perhaps a follow-up issue considering time is important now